crates/re_renderer/src/wgpu_resources/dynamic_resource_pool.rs

```diff
@@ -276,93 +266,108 @@ mod tests {
             assert!(called_destroy);
-            let drop_counter_now = drop_counter.load(Ordering::Acquire);
+            let drop_counter_now = DROP_COUNTER.with(|c| c.get());
             assert_eq!(
                 drop_counter_before + initial_resource_descs.len() * 2,
                 drop_counter_now
             );
             assert_eq!(pool.total_resource_size_in_bytes(), 0);
         }
 
-        // Holding on to a handle avoids both re-use and dropping.
+        // Holding on to the resource avoids both re-use and dropping.
         {
-            let drop_counter_before = drop_counter.load(Ordering::Acquire);
-            let handle0 = pool.alloc(&ConcreteResourceDesc(0), |_| ConcreteResource {
-                drop_counter: drop_counter.clone(),
-            });
-            let handle1 = pool.alloc(&ConcreteResourceDesc(0), |_| ConcreteResource {
-                drop_counter: drop_counter.clone(),
-            });
-            assert_ne!(handle0.handle, handle1.handle);
-            drop(handle1);
+            let drop_counter_before = DROP_COUNTER.with(|c| c.get());
+            let resource0 = pool.alloc(&ConcreteResourceDesc(0), |_| ConcreteResource);
+            let resource1 = pool.alloc(&ConcreteResourceDesc(0), |_| ConcreteResource);
+            assert_ne!(resource0.handle, resource1.handle);
+            drop(resource1);
 
             let mut called_destroy = false;
             pool.begin_frame(4, |_| called_destroy = true);
             assert!(!called_destroy);
-            assert_eq!(drop_counter_before, drop_counter.load(Ordering::Acquire),);
+            assert_eq!(drop_counter_before, DROP_COUNTER.with(|c| c.get()),);
             pool.begin_frame(5, |_| called_destroy = true);
             assert!(called_destroy);
-            assert_eq!(
-                drop_counter_before + 1,
-                drop_counter.load(Ordering::Acquire),
-            );
+            assert_eq!(drop_counter_before + 1, DROP_COUNTER.with(|c| c.get()),);
+        }
+
+        // Two resources have two different handles.
+        {
+            let handle0 = pool
+                .alloc(&ConcreteResourceDesc(0), |_| ConcreteResource)
+                .handle;
+            let handle1 = pool
+                .alloc(&ConcreteResourceDesc(0), |_| ConcreteResource)
+                .handle;
+            assert_ne!(handle0, handle1);
+            pool.begin_frame(5, |_| {});
+        }
+
+        // A resource gets the same handle when re-used.
+        // (important for BindGroup re-use!)
+        {
+            let handle0 = pool
+                .alloc(&ConcreteResourceDesc(0), |_| ConcreteResource)
+                .handle;
+            pool.begin_frame(7, |_| {});
+            let handle1 = pool
+                .alloc(&ConcreteResourceDesc(0), |_| ConcreteResource)
+                .handle;
+            assert_eq!(handle0, handle1);
+            pool.begin_frame(5, |_| {});
```

Comment:
I'm extremely confused by what's going on in this test:
- Are the out of order frame indices relevant? I don't think so?
- Why are these two handles the same? Is it because the test implicitly drops the `Arc` that wraps everything by copying `.handle` out? If so, an explicit `drop` would make it much clearer what's going on

Reply:
> Are the out of order frame indices relevant? I don't think so?

irrelevant. Let me put in some more obvious non-sensical value in there.

> Why are these two handles the same? 

your asessment is correct. I'll make it more obvious!